dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -237,6 +347,9 @@ def _update_categories(
         self._categories = utils.add_nested_dictionaries(
             self._categories, category_count
         )
+        self.update_stop_condition(df_series)
+        if self._stop_condition_is_met:
+            self._categories = {}
```

Comment:
if we already set:
```python
self._stopped_at_unique_ratio = merged_unique_ratio
self._stopped_at_unique_count = merged_unique_count
```
below, we can join it with:
```python
self._categories = {}
```

However, let's consider the function description and the action of the function.

When we say `update_stop_condition`, to me it would only update `_stop_condition_is_met ` boolean. Do we wnat to rename it to be more explicit about it doing more? or do we need `update_stop_condition`?

Could we just have `_update_categories` and `_check_stop_condition_is_met`?

Reply:
Want to note these are code readability / quality comments, not functional as I think the code is functional in this condition.

Reply:
Since this is a feature branch, we can refactor in subsequent PR.